### DISCUSSION TITLE
Using props to pass in an initial value to a ref

### DISCUSSION THREAD

**Discussion by dZlatanovski** [vuejs/docs#2464] [Q&A] [answered]
In the documentation there's an example where a prop is being used to pass in an initial value to a ref: [One way data flow](https://vuejs.org/guide/components/props.html#one-way-data-flow)
I've been using this method to initialise refs in many of my components, but one day my VSCode started showing me an error from eslint and the name of the rule that triggers it is `vue/no-setup-props-destructure`
The description of the rules is as follows:
_This rule reports the destructuring of props passed to setup causing the value to lose reactivity._
I am using Vue3 with SFC and setup scripts along with TypeScript and my eslint configuration is set to use `plugin:vue/vue3-strongly-recommended` .
I have no idea why eslint was okay with this so far, and I am not sure as to why it suddenly started picking this up now, as I am not really destructuring a prop. I tried this in a completely new SFC and even without TypeScript but I am still getting the error. Are the docs wrong? Or is this an eslint problem?
Any explanation would be greatly appreciated.

**Comment by brc-dd** [comment, marked as answer]
https://github.com/vuejs/eslint-plugin-vue/issues/2259
